sdks/java/io/amazon-web-services/src/main/java/org/apache/beam/sdk/io/aws/sns/PublishResultCoder.java

```diff
@@ -42,12 +50,27 @@ static PublishResultCoder of() {
   public void encode(PublishResult value, OutputStream outStream)
       throws CoderException, IOException {
     StringUtf8Coder.of().encode(value.getMessageId(), outStream);
+    StringUtf8Coder.of().encode(value.getSdkResponseMetadata().getRequestId(), outStream);
+    MapCoder.of(StringUtf8Coder.of(), StringUtf8Coder.of())
+        .encode(value.getSdkHttpMetadata().getHttpHeaders(), outStream);
+    VarInt.encode(value.getSdkHttpMetadata().getHttpStatusCode(), outStream);
```

Comment:
nit: I will prefer the HTTP status code to be first just for consistency on the way HTTP works.